UnityProject/Assets/Scripts/Machines/Microwave.cs

```diff
@@ -124,30 +124,50 @@ public void RpcStartCooking()
 	private void FinishCooking()
 	{
 		spriteRenderer.sprite = SPRITE_OFF;
-		if (isServer)
+
+		if (isServer && mealCount > 0)
 		{
 			GameObject mealPrefab = CraftingManager.Meals.FindOutputMeal(meal);
-			SpawnResult result = Spawn.ServerPrefab(mealPrefab, GetComponent<RegisterTile>().WorldPosition, transform.parent);
+			Vector3Int spawnPosition = GetComponent<RegisterTile>().WorldPosition;
 
-			//If the resulting meal has a stackable component, set the amount to mealCount to ensure that food in = food out.
+			SpawnResult result = Spawn.ServerPrefab(mealPrefab, spawnPosition, transform.parent);
 			Stackable stck = result.GameObject.GetComponent<Stackable>();
 
-			if (stck != null && mealCount != 0)
+			if (stck != null)   // If the meal has a stackable component, set the correct number of meals
 			{
-				//Get difference between new item's initial amount and the amount held by mealCount (amount of ingredient).
-				int stckChanger = mealCount-stck.Amount;
+				int spawnedSoFar = stck.InitialAmount;
 
-				//If stckChanger is 0, do nothing.
-				//If stckChanger is positive, add to stack.
-				if (stckChanger > 0)
+				while (spawnedSoFar < mealCount)
 				{
-					stck.ServerIncrease(stckChanger);
-				} else if (stckChanger < 0)
+					int spaceInStack = stck.MaxAmount - stck.Amount;
+
+					if (spaceInStack > 0)
+					{
+						// Add all the missing meals to the stack, or as many as the stack can contain
+						int mealDeficit = mealCount - spawnedSoFar;
```

Comment:
I think Stackable has a "CanAccomodate" method which might be able to take care of some of this logic for you?

Reply:
CanAccomodate only deals with whether two items can exchange stacks, doesn't work for a single object or deal with remaining space.

Reply:
could you add it to can accommodate?
idk if it would cause funny things though

Reply:
It's not something I'd add there, that method deals with unrelated things
I've adjusted Stackable further to address this though